portfolio/src/main/java/com/google/sps/data/BookResponseParser.java

```diff
@@ -0,0 +1,177 @@
+    ArrayList<String> tempAuthors = new ArrayList<String>();
+
+    Book.Builder builder = Book.builder();
+
+    tempIsbn = getIsbn(jsonObject).get(ISBN);
+    builder.isbn(tempIsbn);
+
+    try {
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(TITLE)) {
+        tempTitle = jsonObject.getJSONObject(VOLUME_INFO).get(TITLE).toString();
+        builder.title(tempTitle);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(LANGUAGE)) {
+        tempLang = jsonObject.getJSONObject(VOLUME_INFO).get(LANGUAGE).toString();
+        builder.language(tempLang);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(DESCRIPTION)) {
+        tempDesc = jsonObject.getJSONObject(VOLUME_INFO).get(DESCRIPTION).toString();
+        builder.description(tempDesc);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(INFO_LINK)) {
+        tempInfoLink = jsonObject.getJSONObject(VOLUME_INFO).get(INFO_LINK).toString();
+        builder.infoLink(tempInfoLink);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(PAGE_COUNT)) {
+        tempPageCount =
+            Integer.parseInt(jsonObject.getJSONObject(VOLUME_INFO).get(PAGE_COUNT).toString());
+        builder.pageCount(tempPageCount);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(PUBLISHED_DATE)) {
+        tempPubDate = jsonObject.getJSONObject(VOLUME_INFO).get(PUBLISHED_DATE).toString();
+        builder.publishedDate(tempPubDate);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(PUBLISHER)) {
+        tempPublisher = jsonObject.getJSONObject(VOLUME_INFO).get(PUBLISHER).toString();
+        builder.publisher(tempPublisher);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(MATURITY_RATING)) {
+        tempMatRate = jsonObject.getJSONObject(VOLUME_INFO).get(MATURITY_RATING).toString();
+        builder.maturityRating(tempMatRate);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(CATEGORIES)) {
+        tempCategories =
+            jsonArrayToStringArray(jsonObject.getJSONObject(VOLUME_INFO).getJSONArray(CATEGORIES));
+        builder.categories(tempCategories);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(AUTHORS)) {
+        tempAuthors =
+            jsonArrayToStringArray(jsonObject.getJSONObject(VOLUME_INFO).getJSONArray(AUTHORS));
+        builder.authors(tempAuthors);
+      }
+
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(IMAGE_LINKS)
+          && jsonObject.getJSONObject(VOLUME_INFO)
+                 .getJSONObject(IMAGE_LINKS)
+                 .has(THUMBNAIL)) { // if imageLinks and the thumbnail are present
+        tempThumbnail = jsonObject.getJSONObject(VOLUME_INFO)
+                            .getJSONObject(IMAGE_LINKS)
+                            .get(THUMBNAIL)
+                            .toString();
+        builder.thumbnail(tempThumbnail);
+      }
+    } catch (JSONException e) {
+      e.printStackTrace();
```

Comment:
It might be better to throw this exception. `e.printStackTrace()` will print the exception on the server, and will make it difficult to debug. By throwing exception, you will be able to see the exception being thrown (Please update at other places as well).